### ISSUE TITLE
REQUEST: Jellyfin Client

### ISSUE THREAD

**Issue by phere-x** [m1k1o/neko-apps#12]
https://github.com/jellyfin/jellyfin-media-player

I have all my media organised with my subtitles already on a Jellyfin instance. A lot prefer JF over VLC to browse & manage media already. 

Having something in a native app opposed to it opening in a browser is much more preferable. Can also play wider range of media without needing encoding.

**Comment by m1k1o** [comment]
Looks interesting, lets see how easy it can be built.

**Comment by m1k1o** [comment]
Tried build script from their page:

```
ARG BASE_IMAGE=m1k1o/neko:base
FROM $BASE_IMAGE

#
# build mpv and jellyfin-media-player
RUN set -eux; apt-get update; \
    apt-get install -y --no-install-recommends openbox pip build-essential autoconf automake libtool libharfbuzz-dev libfreetype6-dev libfontconfig1-dev libx11-dev libxrandr-dev libvdpau-dev libva-dev mesa-common-dev libegl1-mesa-dev yasm libasound2-dev libpulse-dev libuchardet-dev zlib1g-dev libfribidi-dev git libgnutls28-dev libgl1-mesa-dev libsdl2-dev cmake wget python g++ qtwebengine5-dev qtquickcontrols2-5-dev libqt5x11extras5-dev libcec-dev qml-module-qtquick-controls qml-module-qtwebengine qml-module-qtwebchannel qtbase5-private-dev curl unzip; \
    pip install meson; \
    mkdir ~/jmp; cd ~/jmp; \
    git clone --depth 1 --branch mpv-build-rewrite https://github.com/avih/mpv-build.git; \
    cd mpv-build; \
    echo --enable-libmpv-shared > mpv_options; \
    echo --disable-cplayer >> mpv_options; \
    echo --disable-gl >> mpv_options; \
    ./rebuild -j4; \
    ./install; \
    ldconfig; \
    cd ~/jmp/; \
    git clone --depth 1 https://github.com/jellyfin/jellyfin-media-player.git; \
    cd jellyfin-media-player; \
    ./download_webclient.sh; \
    cd build; \
    cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/usr/local/ ..; \
    make -j4; \
    make install; \
    rm -rf ~/jmp/; \
    #
    # clean up
    apt-get clean -y; \
    rm -rf /var/lib/apt/lists/* /var/cache/apt/* /tmp/*;

#
# copy configuation files
COPY supervisord.conf /etc/neko/supervisord/jellyfin.conf
COPY openbox.xml /etc/neko/openbox.xml
```

But it fails with `No OpenGL video output found or enabled. Aborting. If you really mean to compile without OpenGL video outputs use --disable-gl.` Even when i added that flag.

**Comment by m1k1o** [comment]
Or rather build browser with built in h265 (or orher codecs) support. I think ungoogled-chromium has that. Or at least had in he future.

**Comment by phere-x** [comment]
That could be an avenue if 265 was the only thing to look for. I think when I was messing around with VLC, performance wise it was much lower than a browser rendering and also Jellyfin running (on the same server). Even with x264/browser playable media.

**Comment by phere-x** [comment]
Saw your Kodi image & I remembered Kodi can integrate with Jellyfin. So installed & followed the steps [here](https://jellyfin.org/docs/general/clients/kodi/). Worked nicely. Has a nice "TV" like interface. JF still looks better & works smoother though.

It's a bit too "open" for me. I'd want to only display library & give access to nothing else. So I could let whoever request for stuff to queue and/or have remote control capability. But maybe I'm wanting a whole new different app 😅

I think integrating/developing a new app like this for Neko would fit in really well. If you're willing to hear & explore it. Either something utilizing JF/Sonarr/Radarr api to grab movie data. Or maybe a more lightweight tool that scans directories to find & identify media. Have a tab on chat that lets viewers browse & request/vote media to be played.

One reason why I wanted MPV was to utilize its RC feature to send commands to a running instance. So I can have something listen available local media (or search for it on some sites & grab `.m3u8`), then queue it

**Comment by m1k1o** [comment]
Yes, that looks like a whole new different app, but this usecase with necko could fit *arr stack. If you are willing to take this and develop this (either on top of neko or standalone), or find maintainer for this project, that would be great.

Or maybe there is a way how to customize KODI for this usecase.

**Comment by phere-x** [comment]
I would probably mess around with it & make something very basic. Hopefully someone could help/flesh it out if I can.

I know kodi has JSON-RPC capabilities. I don't know much about their API & what info could be requested in terms of available media.

With *arr apps that is quite easy. But everyone might not want to install 2 independent apps for this. I remember there was an "auto-organising" app to properly format & order one's collection so maybe that would be an option instead. If I can remember I'll look into that.

**Comment by m1k1o** [comment]
Here is my attempt. https://github.com/m1k1o/neko-apps/tree/jellyfin
It fails somewhere. If you have time to debug, you can take a look at that.
